pyam/core.py

```diff
@@ -405,6 +405,28 @@ def validate(self, criteria={}, exclude=False):
 
             return df
 
+    def conv_unit(self, unit_conv, inplace=False):
+        """ Converts units based on provided unit conversion factors
+
+        Parameters
+        ----------
+        unit_conv: dict
+            for each unit for which a conversion should be carried out, provide current unit and target unit and conversion factor
+            {<current unit>: [<target unit>, <conversion factor>]}
+        inplace: bool, default False
+            if True, do operation inplace and return None
+        """
+        ret = copy.deepcopy(self) if not inplace else self
+        for unt in unit_conv:
+            if not type(unit_conv[unt][1]) in [int, np.integer, float]:
+                raise ValueError(
+                    'unit conversion factor for {} is not numeric!'.format(unt))
+            ret.data.loc[ret.data['unit'] == unt, 'value'] = ret.data.loc[ret.data['unit']
```

Comment:
the expression `ret.data['unit'] == unt` is used three times, better to starts with 

```
rows = ret.data['unit'] == unt
```

and then use `rows` in the subsequent two lines?